test/cache_consistency_test.rb

```diff
@@ -0,0 +1,20 @@
+require "test_helper"
+
+# In general we do not want to expose the innter workings of the caching and unit systems as part of the API.
+# But we do want to be able to test that there is no conflict between what's in the cache file and what is defined
+# in the unit systems. So we compromise by reaching into implementation to force compare them.
```

Comment:
Testing compromises here too.

Had to reach into internals to compare the expected and actual. But I'm ok with this because exposing these internals in the API just for testing is worse than having to update this test later.

Reply:
This test will fail if you modify the measurement system without updating the cache.